docs/snippets/common/ghp-github-action.js.mdx

```diff
@@ -0,0 +1,26 @@
+```shell
+# .github/workflows/deploy-ghp.yml
+
+# Workflow name
+name: Build and Publish storybook to GitHub Pages
+
+on:
+  # Event for the workflow
+  push:
+    branches:
+      - "main"
+
+# List of jobs
+jobs:
+  deploy:
+    # Operating System
+    runs-on: ubuntu-latest
+
+    # Job steps
+    steps:
+
+    #👇 Adds Storybook build and deploy to GitHub Pages as a step in the workflow
+    - uses: bitovi/github-actions-storybook-to-github-pages@v1.0.1
+      with:
+        path: dist/storybook
+```
```

Comment:
@mickmcgrath13 I'd like to know if we can improve the example and provide a more complete workflow. We could turn this into the following:

```yaml
# .github/workflows/github-pages-deploy.yml

# Workflow name
name: Build and Publish Storybook to GitHub Pages

on:
  # Event for the workflow to run on
  push:
    branches:
      - "your-branch-name" # Replace with the branch you want to deploy from

permissions:
  contents: read
  pages: write
  id-token: write

# List of jobs
jobs:
  deploy:
    runs-on: ubuntu-latest
    # Job steps
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:
          node-version: "16.x"
      #👇 Adds Storybook build and deploy to GitHub Pages as a step in the workflow
      - uses: bitovi/github-actions-storybook-to-github-pages@v1.0.1
        with:
          install_command: yarn install
          build_command: yarn build-storybook
          path: storybook-static
          checkout: false
```
The reasoning behind this is that we're indirectly mentioning the available options and required configuration for permissions to allow a user to copy and paste, and save it in their environment, and deploy it without running into issues. Let me know what you think, and we'll go from there.

Reply:
Hey @jonniebigodes, great suggestion!

I've updated the docs as suggested and added a few more comments to further help users